datadog/resource_datadog_dashboard.go

```diff
@@ -4641,11 +4694,134 @@ func buildTerraformQueryTableRequests(datadogQueryTableRequests *[]datadogV1.Tab
 			}
 			terraformRequest["cell_display_mode"] = terraformCellDisplayMode
 		}
+		if v, ok := datadogRequest.GetTextFormatsOk(); ok {
+			terraformTextFormats := make([][]map[string]interface{}, len(*v))
+			for i, textFormat := range *v {
+				test := buildTerraformQueryTableTextFormat(&textFormat)
+				terraformTextFormats[i] = *test
+			}
+			terraformRequest["text_formats"] = terraformTextFormats
+		}
 		terraformRequests[i] = terraformRequest
 	}
 	return &terraformRequests
 }
 
+// Query Table Widget Text Format Helpers
+func buildDatadogQueryTableTextFormat(terraformQueryTableTextFormat *[]interface{}) *[]datadogV1.TableWidgetTextFormatRule {
+	datadogQueryTableTextFormat := make([]datadogV1.TableWidgetTextFormatRule, len(*terraformQueryTableTextFormat))
+	for j, textFormatRule := range *terraformQueryTableTextFormat {
+		terraformTextFormatRule := textFormatRule.(map[string]interface{})
+		terraformTextFormatMatch := terraformTextFormatRule["match"].(map[string]interface{})
+		datadogMatch := datadogV1.NewTableWidgetTextFormatMatch(datadogV1.TableWidgetTextFormatMatchType(terraformTextFormatMatch["type"].(string)), terraformTextFormatMatch["value"].(string))
+		datadogTextFormatRule := datadogV1.NewTableWidgetTextFormatRule(*datadogMatch)
+		// Optional
+		if v, ok := terraformTextFormatRule["replace"].(map[string]interface{}); ok {
+			if w, ok := v["type"].(string); ok && len(w) != 0 {
+				switch w {
+				case "all":
+					datadogReplace := datadogV1.NewTableWidgetTextFormatReplaceAll(datadogV1.TABLEWIDGETTEXTFORMATREPLACEALLTYPE_ALL, v["with"].(string))
+					datadogTextFormatRule.SetReplace(datadogV1.TableWidgetTextFormatReplaceAllAsTableWidgetTextFormatReplace(datadogReplace))
+				case "substring":
+					datadogReplace := datadogV1.NewTableWidgetTextFormatReplaceSubstring(v["substring"].(string), datadogV1.TABLEWIDGETTEXTFORMATREPLACESUBSTRINGTYPE_SUBSTRING, v["with"].(string))
+					datadogTextFormatRule.SetReplace(datadogV1.TableWidgetTextFormatReplaceSubstringAsTableWidgetTextFormatReplace(datadogReplace))
+				}
+			}
+		} else {
+			datadogTextFormatRule.Replace = nil
```

Comment:
Probably not necessary but I tried with and without this expression (and those below) and got the same result.